xla/service/gpu/fusions/transpose_test.cc

```diff
@@ -246,6 +246,71 @@ TEST_F(TransposeTest, ThreadIndexingPartialBlock) {
       )"));
 }
 
+TEST_F(TransposeTest, ThreadIndexingSidedOutput) {
+  auto module = ParseAndReturnVerifiedModule(R"(
+    HloModule module
+
+    fusion {
+      %input = f32[100,32,64] parameter(0)
+      %bitcast0 = f32[100,64,32] bitcast(%input)
+      %transpose = f32[100,64,32] transpose(%input), dimensions={0,2,1}
+      %bitcast1 = f32[100,2048] bitcast(%bitcast0)
+      ROOT %tuple = (f32[100,64,32], f32[100,2048]) tuple(%transpose, %bitcast1)
+    }
+
+    ENTRY entry {
+      %input = f32[100,32,64] parameter(0)
+      ROOT %fusion = (f32[100,64,32], f32[100,2048]) fusion(%input), kind=kInput, calls=fusion
+    })")
+                    .value();
+
+  auto* root = module->entry_computation()->root_instruction();
+  auto analysis = AnalyzeFusion(*root, device_info_);
+
+  TF_ASSERT_OK_AND_ASSIGN(auto fusion, GetTransposeFusion(analysis));
+  mlir::MLIRContext mlir_context;
+
+  EXPECT_THAT(
+      fusion->ComputeThreadIdToInputIndexing(1, 0, &mlir_context)->ToString(),
```

Comment:
Let's add a test that verifies that the input indexing for root 0 is the same as the input indexing for root 1 (composed with %bitcast0, so we get the indexing for %input).

Reply:
They are not same in my opinion. The input indexing of root0 corresponds to [100, 32, 64] but root1 corresponds to [100, 64, 32]. If my understanding is wrong, please correct me.

Reply:
Unless I'm missing something, the test compares the input indexing for root 1 to the output indexing for root 1. We should compare the input indexing for root 0 and 1.

Reply:
I think you may misunderstand something. Line 273 compare the expected input indexing for root1 to actual input indexing for root1. If compare to output indexing for root1, the size of affine map results should be 3 not 2, can see Line 276.

Reply:
My comment was only about **input** indexing, not about outputs.

Reply:
I may understand what you mean. You advice me add test to check input indexing for root0 in order to ensure ComputeThreadIdToInputIndexing func can get correct root0 input indexing result for the tested hlo with side output, right?

Reply:
Yes, we should verify that the input for the transpose and the side output have the same indexing (so the reads will be shared).

Reply:
You can see if this test as you expected. https://github.com/openxla/xla/pull/12297/files#diff-77c8f926d853dcddb6d390925c7dd4bcc05f4a811a4cd6a3946689b3c3b58683R249